src/main/java/com/maxmind/geoip2/model/AbstractCityResponse.java

```diff
@@ -36,7 +37,27 @@
         this.subdivisions = subdivisions != null ? subdivisions : new ArrayList<>();
     }
 
-    protected static ArrayList<Subdivision> mapSubdivisions(
+    AbstractCityResponse(
+            AbstractCityResponse response,
+            String ipAddress,
+            Network network,
+            List<String> locales
+    ) {
+        this(
+            response.getCity() != null ? new City(response.getCity(), locales) : null,
+            response.getContinent() != null ? new Continent(response.getContinent(), locales) : null,
```

Comment:
Perhaps we could further share code by moving the country stuff to `AbstractCountryResponse` and then call that constructor here.

Reply:
Yeah, true. I thought about it but it didn't seem too big a win. I'll try it though.